src/helpers/MonitoringState.js

```diff
@@ -0,0 +1,104 @@
+import { logger } from "./logger.js";
+import { OracleMonitorConfig } from "./OracleMonitorConfig.js";
+import { saveJSONDataToFile } from "./utils.js";
+
+export class MonitoringState {
+  /**
+   * Constructor for class
+   * @param {string} filePath file path for the state file
+   * @param {OracleMonitorConfig} oracleConfig config
+   */
+  constructor(filePath, oracleConfig) {
+    try {
+      this.stateFile = filePath;
+      this.state = this.readMonitoringState(oracleConfig);
```

Comment:
`this.readMonitoringState` is a void function, so it does not return anything.

```suggestion
      this.readMonitoringState(oracleConfig);
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/268caad22ff73c68b8b647687f00c8725af39911